### ISSUE TITLE
SE_BAD_FIELD with records and spotbugs-4.8.4

### ISSUE THREAD

**Issue by rovarga** [spotbugs/spotbugs#2935]
A simple non-serializable record as available here: https://github.com/opendaylight/yangtools/blob/2e257731e354e6cc2fd45a6f6eec79bd01490c07/parser/yang-parser-api/src/main/java/org/opendaylight/yangtools/yang/parser/api/YangLibModule.java triggers:

```
[ERROR] Low: Class org.opendaylight.yangtools.yang.parser.api.YangLibModule defines non-transient non-serializable instance field source [org.opendaylight.yangtools.yang.parser.api.YangLibModule] In YangLibModule.java SE_BAD_FIELD
[ERROR] Low: Class org.opendaylight.yangtools.yang.parser.api.YangLibModule defines non-transient non-serializable instance field submodules [org.opendaylight.yangtools.yang.parser.api.YangLibModule] In YangLibModule.java SE_BAD_FIELD
```

with spotbugs-4.8.4. No warning is reported with spotbugs-4.8.3.

**Comment by gtoison** [comment]
Does it only happen for these two record fields?
It seems a bit strange indeed that it complains for `@NonNull SourceRepresentation source` but not for `@NonNull SourceIdentifier identifier`
I tried reproducing the problem but no luck so far

**Comment by rovarga** [comment]
Yes, it's only those two, and the reported warnings would be correct, as of all the referenced types are Serializable except SourceRepresentation  and YangLibSubmodule (and YangLibSubmodule is again a record referencing SourceRepresetation -- except neither YangLibModule nor YangLibSubmodule are marked as Serializable, which is very weird.

**Comment by gtoison** [comment]
Hum, there seems to be some kind of heuristic going on, SpotBugs does this test: `if (isSerializable < 0.9)`
I'll try to figure out what this code is doing

**Comment by hazendaz** [comment]
fixed by #2962
